zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -1311,6 +1462,42 @@ pub(crate) fn next_check_time<R: RngCore, D: DerefMut<Target = R>>(
     Ok(from_event + Duration::new(event_delay, 0))
 }
 
+pub(crate) fn schedule_next_check<P: consensus::Parameters, C: Clock, R: RngCore>(
+    conn: &rusqlite::Transaction,
+    params: &P,
+    clock: C,
+    mut rng: R,
+    address: &TransparentAddress,
+    offset_seconds: u32,
+) -> Result<Option<SystemTime>, SqliteClientError> {
+    let addr_str = address.encode(params);
+    let now = clock.now();
+    let next_check = next_check_time(&mut rng, now, offset_seconds)?;
+    let scheduled_next_check = conn
+        .query_row(
+            "UPDATE addresses
+             SET transparent_receiver_next_check_time = CASE
+                WHEN transparent_receiver_next_check_time < :current_time THEN :next_check
```

Comment:
Isn't there a race condition here? The current time being after `transparent_receiver_next_check_time` isn't equivalent to us actually having processed the event that is supposed to be at `transparent_receiver_next_check_time`.